OMCompiler/Compiler/BackEnd/CommonSubExpression.mo

```diff
@@ -671,6 +680,11 @@ algorithm
       algorithm
         globalKnownVarHT := BaseHashSet.add(cr, globalKnownVarHT);
      then ();
+
+     else algorithm
+        Error.addInternalError("addConstantCseVarsToGlobalKnownVarHT failed. Reached else case that should not be reachable while handling CSE expression:\n" + ExpressionDump.dumpExpStr(cse_crExp, 0), sourceInfo());
+        fail();
+      then();
```

Comment:
Not a review per se, but we really need to bite the bullet and add something to MetaModelica that enforces "nofail" functions. Basically automatically adding else cases with errors to match based functions like this. Of course you can always write an else case yourself but it is tedious and easy to overlook. 

It will save us from some debugging effort later.

Reply:
I agree completely. That's why the original author of the problem was only seeing some generic error message
```
post-optimization module wrapFunctionCalls (simulation) failed
```